qiskit_ibm_runtime/estimator.py

```diff
@@ -317,7 +317,14 @@ def _run(  # pylint: disable=arguments-differ
             combined["optimization_level"] = Options._DEFAULT_OPTIMIZATION_LEVEL
             combined["resilience_level"] = Options._DEFAULT_RESILIENCE_LEVEL
         logger.info("Submitting job using options %s", combined)
-        inputs.update(Options._get_program_inputs(combined))
+        if self._session.backend():
+            inputs.update(
+                Options._get_program_inputs(
+                    combined, primitive="Estimator", backend=self._session.backend()
+                )
+            )
+        else:
+            inputs.update(Options._get_program_inputs(combined, primitive="Estimator"))
```

Comment:
`self._session.backend()` would return `None` if there is no backend, right? So there is no need for the if/else.